sanic/response/convenience.py

```diff
@@ -331,3 +346,94 @@ async def _streaming_fn(response):
         headers=headers,
         content_type=mime_type,
     )
+
+
+class AutoIndex:
+    STYLE = """
+        html { font-family: sans-serif; }
+        main { padding: 1rem; }
+        table { width: 100%; max-width: 1200px; }
+        td { font-family: monospace; }
+        table.autoindex td:first-child { width: 65% }
+        table.autoindex td:last-child { text-align: right; width: 15%; }
+        span.icon { margin-right: 1rem; }
+        @media (prefers-color-scheme: dark) {
+            html { background: #111; color: #ccc; }
+            a { color: #ccc; }
+            a:visited { color: #777; }
+        }
+    """
+    TITLE = "📁 File browser"
+
+    def __init__(
+        self, directory: Path, autoindex: bool, index_name: str
+    ) -> None:
+        self.directory = directory
+        self.autoindex = autoindex
+        self.index_name = index_name
+
+    async def handle(self):
+        index_file = self.directory / self.index_name
+        if self.autoindex and (not index_file.exists() or not self.index_name):
+            return await self.index()
+
+        if self.index_name:
+            return await file(index_file)
+
+    async def index(self):
+        return html(self.render())
+
+    def render(self) -> str:
+        doc = Document(title=self.TITLE, lang="en")
+        self._head(doc)
+        with doc.main:
+            self._headline(doc)
+            self._file_table(doc)
+        return str(doc)
+
+    def _head(self, doc: Document):
+        doc.head.title(self.TITLE).style(self.STYLE)
```

Comment:
The `Document` function already adds a title, so the only thing needed in `_head` is the inline style (notice that the `head` tag is also not needed in HTML5). Possibly refactor not to have it broken down to that many methods, only a few bigger parts (that one might wish to override I suppose).

For purposes of typing, you need `html5tagger.Builder`, which is what using `Document` or `E` returns. Possibly use `self.doc` to avoid having to pass it around?

Reply:
I added it explicitly because it was not inside head. When I put the generated code into a validation tool it yelled at me for not having it.